string[] name
Pose2D[] location
